File: udt-c/src-gen-sav/dab/compte.c
#include <dab/compte.h>
#include <stdio.h>

util_error dab_compte_put( dab_compte * This, io_byte_buffer * target ) {
   UTIL_ERROR_CHECK( io_byte_buffer_put_string( target, This->id       ), __FILE__, __LINE__ );
   UTIL_ERROR_CHECK( io_byte_buffer_put_double( target, This->solde    ), __FILE__, __LINE__ );
   UTIL_ERROR_CHECK( io_byte_buffer_put_bool  ( target, This->autorise ), __FILE__, __LINE__ );
   return UTIL_NO_ERROR;
}

util_error dab_compte_get( dab_compte * This, io_byte_buffer * source ) {
   UTIL_ERROR_CHECK( io_byte_buffer_get_string( source, This->id, sizeof( This->id )), __FILE__, __LINE__ );
   UTIL_ERROR_CHECK( io_byte_buffer_get_double( source, &This->solde                ), __FILE__, __LINE__ );
   UTIL_ERROR_CHECK( io_byte_buffer_get_bool  ( source, &This->autorise             ), __FILE__, __LINE__ );
   return UTIL_NO_ERROR;
}

File: udt-java/src/udt/Compte.java
package udt;

class Compte {

   boolean _isValid;
   String  _id;
   double  _solde;
   boolean _autorise;

   Compte() {
      _isValid  = false;
      _id       = "";
      _solde    = 0.0;
      _autorise = false;
   }

   void set( String id, double  solde, boolean autorise ) {
      _id       = id;
      _solde    = solde;
      _autorise = autorise;
      _isValid  = ( _id.length() > 0 )&&( _solde > 0.0 );
   }

   public void set( dab.Compte compte ) {
      _id       = compte.id;
      _solde    = compte.solde;
      _autorise = compte.autorise;
      _isValid  = ( _id.length() > 0 )&&( _solde > 0.0 );
   }

   void invalidate() {
      System.err.print( "Compte.invalidate\n" );
      _isValid = false;
   }

   boolean isValid() { return _isValid; }

   String getId() { return _id; }

   double  getSolde() { return _solde; }
}